### PR TITLE
Compiled the full recipe list

### PR DESCRIPTION
**Pull request by theis999**
Updated the full list of recipes with the help of lua code.

### CODE REVIEW

**Comment by MortenTobiasNielsen** [comment]
Very nice - we will need to ensure that the added recipes are also available as buildings. It might be a little frustration that some things cannot be crafted, but at least it is possible to place everything which can be crafted. :D

**Comment by theis999** [comment]
Looking over it again i can see i am missing comma between ingredient items, so i will have to fix that.
I can also see that the list includes loaders and rocket parts, so i will remove those.

Does that cover your concerns?

**Comment by MortenTobiasNielsen** [comment]
I think I would like to save this for the C# version, I would rather spend the mental capacity testing it in that context. :)

**Comment by theis999** [comment]
@MortenTobiasNielsen I agree

**Comment by MortenTobiasNielsen** [comment]
> Looking over it again i can see i am missing comma between ingredient items, so i will have to fix that. I can also see that the list includes loaders and rocket parts, so i will remove those.
> 
> Does that cover your concerns?

If you fix those problems, then I will try to merge it ins and see if it causes problems. :)

**Comment by theis999** [comment]
@MortenTobiasNielsen done!

File: src/utils.h [hunk 1]
```diff
@@ -78,62 +78,216 @@ static const std::vector<std::string> splitter_list = { "Splitter",
...
-																		 { "Logistic science pack", {"Inserter", "1", "Transport belt", "1" }},
-																		 { "Chemical science pack", {"Advanced circuit", "3", "Engine unit", "2", "Sulfur", "1" }},
-																		 { "Production science pack", {"Electric furnace", "1", "Productivity module", "1", "Rail", "30" }},
-																		 { "Utility science pack", {"Flying robot frame", "1", "Low density structure", "3", "Processing unit", "2" }},
-																	   };
+static const std::map<std::string, std::vector<std::string>> recipes = { 
+	{ "Assembling machine 1", { "Iron plate", "9" , "Iron gear wheel", "5" , "Electronic circuit", "3"  }},
+	{ "Automation science pack", { "Copper plate", "1" , "Iron gear wheel", "1"  }},
+	{ "Firearm magazine", { "Iron plate", "4"  }},
+	{ "Pistol", { "Iron plate", "5" , "Copper plate", "5"  }},
```
Comment (MortenTobiasNielsen) [inline src/utils.h:85]: Sorry for the late reply. There is a space after the amount, which I don't think should be there, so it is consistent with the spacing of the name.

File: src/utils.h [hunk 2]
```diff
@@ -78,62 +78,216 @@ static const std::vector<std::string> splitter_list = { "Splitter",
...
+	{ "Small lamp", { "Iron plate", "1" , "Copper cable", "3" , "Electronic circuit", "1"  }},
+	{ "Basic oil processing", { "Crude oil", "100"  }},
+	{ "Advanced oil processing", { "Water", "50" , "Crude oil", "100"  }},
+	{ "Coal liquefaction", { "Coal", "10" , "Heavy oil", "25" , "Steam", "50"  }},
+	{ "Pipe", { "Iron plate", "1"  }},
+	{ "Pipe to ground", { "Iron plate", "5" , "Pipe", "10"  }},
+	{ "Logistic robot", { "Advanced circuit", "2" , "Flying robot frame", "1"  }},
+	{ "Construction robot", { "Electronic circuit", "2" , "Flying robot frame", "1"  }},
+	{ "Energy shield equipment", { "Steel plate", "10" , "Advanced circuit", "5"  }},
+	{ "Energy shield mk2 equipment", { "Processing unit", "5" , "Low density structure", "5" , "Energy shield equipment", "10"  }},
```
Comment (MortenTobiasNielsen) [inline src/utils.h:117]: I think all of the "equipment" need to have a conversion, The name in the game is Energy shield MK2 and I have made the item list like the once in the game, so when I chose to set the recipe for Energy shield MK2 it isn't able to find the recipe. You might be able to use the "GenerateScript::check_item_name" method for the conversion.

**Review by MortenTobiasNielsen — COMMENTED** [review]
Looks good, but I have two comments.

File: src/utils.h (hunk 1 above)
Comment (theis999) [reply src/utils.h]: I ran it through a format tool, so it should be alright now.

File: src/utils.h (hunk 2 above)
Comment (theis999) [reply src/utils.h]: [Energy_shield_MK2](https://wiki.factorio.com/Energy_shield_MK2)
The external english name is `Energy shield MK2` but the internal name is `energy-shield-mk2-equipment` both of the item and the recipe (and technology). 
If we need to make many conversions, then we should have a conversion table.

Comment (MortenTobiasNielsen) [reply src/utils.h]: I haven't run into anything else than the equipment

**Comment by theis999** [comment]
Changed the names of equipments

**Comment by MortenTobiasNielsen** [comment]
I am pretty sure that won't work - have you tested it?

**Comment by theis999** [comment]
It does work, in my tests

**Comment by MortenTobiasNielsen** [comment]
I get this error:
image

**Comment by theis999** [comment]
I don't understand why mk2 should be lowercase, but ok

**Comment by MortenTobiasNielsen** [comment]
I will merge it in and make the changes on my testing branch.

**Pull request merged**